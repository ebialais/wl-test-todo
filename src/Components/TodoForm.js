import React, { useState } from 'react';
import './style.css';

const TodoForm = (props) => {
    const [input, setInput] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault();
        props.addTodo(input)
        setInput('');
    };

    const handleChange = (e) => {
        setInput(e.target.value)
    };

    return (
        <div className='todo-form-container'>
            <form className='todo-form' onSubmit={handleSubmit}>
                <input
                    className='todo-input'
                    type='text'
                    data-testid='todoForm-input'
                    placeholder='Add something to do'
                    value={input}
                    onChange={handleChange}
                />
                <button className='todo-form-button'>Add</button>
            </form>
        </div>
    )
};

export default TodoForm;
