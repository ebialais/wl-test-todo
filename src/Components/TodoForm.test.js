import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import user from '@testing-library/user-event';
import TodoForm from './TodoForm';


it('should accept value for new todo', () => {
    const mockAddTodo = jest.fn()
    const { getByRole, getByText } = render( <TodoForm addTodo={mockAddTodo}/> )

    const input = getByRole('textbox', {name : ''});
    user.type(input, 'a new todo');
    expect(input.value).toEqual('a new todo');

    const submitButton = getByText(/add/i);
    fireEvent.click(submitButton);
    expect(mockAddTodo).toHaveBeenCalledTimes(1);
    
});

test('test todoForm rendering', async () => {
    const mockAddTodo = jest.fn()

    await act(async () => {
        await act(async () => {
            const renderer = new ShallowRenderer();
            const renderedTodoForm = renderer.render( <TodoForm addTodo={mockAddTodo} /> );
            expect(renderedTodoForm).toMatchSnapshot('rendered TodoForm');
        });
    });
});