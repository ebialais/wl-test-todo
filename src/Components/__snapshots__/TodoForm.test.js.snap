// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test todoForm rendering: rendered TodoForm 1`] = `
<div
  className="todo-form-container"
>
  <form
    className="todo-form"
    onSubmit={[Function]}
  >
    <input
      className="todo-input"
      onChange={[Function]}
      placeholder="Add something to do"
      type="text"
      value=""
    />
    <button
      className="todo-form-button"
    >
      Add
    </button>
  </form>
</div>
`;
